libsolidity/parsing/Parser.cpp

```diff
@@ -426,12 +437,13 @@ ASTPointer<ASTNode> Parser::parseFunctionDefinitionOrFunctionTypeStateVariable(A
 		}
 		else
 			m_scanner->next(); // just consume the ';'
-		bool const c_isConstructor = (_contractName && *header.name == *_contractName);
+		if (_contractName && *header.name == *_contractName)
+			header.isConstructor = true;
```

Comment:
Since this changes the `header`, I'd move to this to be local in `parseFunctionHeader`.